### PR TITLE
feat: add testToken

### PR DESCRIPTION
**Pull request by dimsome**

### CODE REVIEW

File: deployments/supertoken/prod_testing_mainnet_addresses.json
Comment (tHeMaskedMan981) [inline deployments/supertoken/prod_testing_mainnet_addresses.json:1]: we want to keep this file?

File: script/helpers/verifyConstants.ts [hunk 1]
```diff
@@ -32,16 +32,33 @@ export const verifyConstants = async () => {
...
-      checkMissingFields({
-        name,
-        symbol,
-        decimals,
-        initialSupplyOwner,
-        owner,
-        initialSupply,
-      });
+
+      if ("address" in superTokenInfo) {
```
Comment (tHeMaskedMan981) [inline script/helpers/verifyConstants.ts:36]: are we planning to keep address in superTokenInfo object? i believe would be better to keep all the addresses at one place in existing-token-addresses file.

File: src/enums/projectType.ts [hunk 2]
```diff
@@ -13,4 +13,12 @@ export const ProjectTypeMap: Record<Project, ProjectType> = {
   [Project.SPECTRAL_SIGNAL]: ProjectType.SUPERTOKEN,
   [Project.LEAF_TESTNET]: ProjectType.SUPERTOKEN,
   [Project.MIST_TESTNET]: ProjectType.SUPERTOKEN,
+  [Project.SYNDR_SEPOLIA_TESTNET]: ProjectType.SUPERBRIDGE,
+  [Project.TESTING_MAINNET]: ProjectType.SUPERTOKEN,
```
Comment (tHeMaskedMan981) [inline src/enums/projectType.ts:17]: can cleanup here as well, testing projects.

File: script/setup.ts [hunk 3]
```diff
@@ -1,5 +1,8 @@
 import { setupConfigs } from "./setup/setupConfig";
 
+import { EventEmitter } from "events";
+EventEmitter.defaultMaxListeners = 20;
```
Comment (tHeMaskedMan981) [inline script/setup.ts:4]: adding this removed the warning you were getting while running scripts ?

File: script/setup/common.ts [hunk 4]
```diff
@@ -15,15 +15,15 @@ export type ProjectConfig = {
...
+  newToken?: boolean;
 };
 
 export type NewTokenInfo = {
-  name: string;
-  symbol: string;
-  decimals: number;
-  chainSlug: ChainSlug;
   address: string;
+  name?: string;
```
Comment (tHeMaskedMan981) [inline script/setup/common.ts:23]: why are we making these optional? want to understand the flow better

File: script/setup.ts (hunk 3 above)
Comment (dimsome) [reply script/setup.ts:4]: yes, I have not had any logs related to EventEmitter after

File: script/helpers/verifyConstants.ts (hunk 1 above)
Comment (dimsome) [reply script/helpers/verifyConstants.ts:36]: The idea was here that if we supply a token, then we won't need to supply any other information. Hence the token already exists and we don't need to deploy anymore

File: script/setup/common.ts (hunk 4 above)
Comment (dimsome) [reply script/setup/common.ts:23]: I had before a interface where you would either supply an address, then nothing else, or everything else but not the address. To define whether the token needs to be deployed or is already deployed.

**Review by dimsome — COMMENTED** [review]
ok i think is ready

**Pull request merged**